<!DOCTYPE html>
<html lang="en"><head>
 <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin Jekyll SEO tag v2.8.0 -->
<title>MetaWorks | A multi-marker metabarcode bioinformatic pipeline</title>
<meta name="generator" content="Jekyll v3.9.2" />
<meta property="og:title" content="MetaWorks" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="A multi-marker metabarcode bioinformatic pipeline" />
<meta property="og:description" content="A multi-marker metabarcode bioinformatic pipeline" />
<link rel="canonical" href="http://localhost:4000/" />
<meta property="og:url" content="http://localhost:4000/" />
<meta property="og:site_name" content="MetaWorks" />
<meta property="og:type" content="website" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="MetaWorks" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebSite","description":"A multi-marker metabarcode bioinformatic pipeline","headline":"MetaWorks","name":"MetaWorks","url":"http://localhost:4000/"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/assets/main.css"><link type="application/atom+xml" rel="alternate" href="http://localhost:4000/feed.xml" title="MetaWorks" /><script src="/assets/javascript/bootstrap/jquery.min.js"></script>
  <script src="/assets/javascript/bootstrap/bootstrap.bundle.min.js"></script>
</head>
<body><nav class="navbar navbar-expand-lg navbar-dark bg-primary">
  <div class="container">
    <a class="navbar-brand" rel="author" href="/">MetaWorks</a><button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent"
      aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav me-auto"><li class="nav-item">
            <a class="nav-link" href="/quickstart/">Quickstart</a>
          </li><li class="nav-item">
            <a class="nav-link" href="/workflows/">Workflows in Detail</a>
          </li><li class="nav-item">
            <a class="nav-link" href="/tutorial/">Tutorial</a>
          </li><li class="nav-item">
            <a class="nav-link" href="/faq/">FAQ</a>
          </li></ul>
    </div>
  </div>
</nav>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        <p align="center">
  <img src="./images/MetaWorksLogo.jpg" width="750" />
</p>

<p><br /></p>
<h1> Why MetaWorks? </h1>

<p><br />MetaWorks comes with a conda environment file MetaWorks_v1.10.0 that should be activated before running the pipeline.  Conda is an environment and package manager (Anaconda, 2016).  The environment file contains most of the programs and dependencies needed to run MetaWorks.  If pseudogene filtering will be used, then the NCBI ORFfinder program will also need to be installed.  Additional RDP-trained reference sets may need to be downloaded if the reference set needed is not already built in to the RDP classifier (see Table 1 below).</p>

<p>Snakemake is a python-based workflow manager (Koster and Rahmann, 2012) and it requires three sets of files to run the any one of the workflows described described in the next section (Fig 1).</p>

<p><strong>Fig 1.  Using a conda environment helps to quickly gather programs and dependencies used by MetaWorks.</strong>.</p>

<p><img src="./images/conda_env.png" width="750" /></p>

<p>The configuration file is edited by the user to specify directory names, indicate the sample and read fields from the sequence filenames, and specify other required pipeline parameters such as primer sequences, marker name, and whether or not pseudogene filtering should be run.</p>

<p>The snakefile describes the pipeline itself and normally does not need to be edited in any way (Fig 2).</p>

<p><strong>Fig 2. The snakefile describes the programs used, the settings, and the order in which to run commands.</strong><br />
The pipeline takes care of any reformatting needed when moving from one step to another.  In addition to the final results file, a summary of statistics and log files are also available for major steps in the pipeline.</p>

<p><img src="./images/dataflow.png" width="600" /></p>

<p>The pipeline begins with raw paired-end Illumina MiSeq fastq.gz files.  Reads are paired.  Primers are trimmed.  All the samples are pooled for a global analysis.  Reads are dereplicated, denoised, and chimeric sequences are removed producing a reference set of denoised exact sequence variants (ESVs). At this step, the pipeline diverges into several paths:  an ITS specific dataflow, a regular dataflow, and a pseudogene filtering dataflow.  For ITS sequences, flanking rRNA gene regions are removed then they are taxonomically assigned.  For the regular pipeline, the denoised ESVs are taxonomically assigned using the RDP classifier.  If a protein coding marker is being processed you have the option to filter out putative pseudogenes (Porter and Hajibabaei, 2021).  The result is a report containing a list of ESVs for each sample, with read counts, and taxonomic assignments with a measure of bootstrap support (Fig 3).</p>

<p><strong>Fig 3. The RDP classifier produces a measure of confidence for taxonomic assignments at each rank.</strong><br />
Results can be filtered by bootstrap support values to reduce false-positive assignments.  The appropriate cutoffs to use for filtering will depend on the marker/classifier used, query length, and taxonomic rank.  See links in Table 1 for classifier-specific cutoffs to ensure 95-99% accuracy.</p>

<p><img src="./images/taxonomic_assignments.png" width="600" /></p>

      </div>
    </main>
<div class="py-5 border-top">
  <div class="container">
  	<div class="col-sm text-right">
    	<p>This site is maintained by Teresita M. Porter (terrimporter AT gmail DOT com) and Artin Mashayekhi</p>
		</p>
      </div>

    </div>

  </div>
</div>
</body>

</html>
